### PR TITLE
Remove c indexing 4

### PR DESCRIPTION
**Pull request by agitator**

### CODE REVIEW

**Comment by agitator** [comment]
This is not removing support for 4.2, but having a test setup running on 3 different plone version is hassle enough.
Check Travis, jobs are running for 4.3, 5.0 and 5.1
I also pushed the version to 2.0

File: .travis.yml [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
 cache:
+  pip: true
   directories:
-  - buildout-cache
+    - eggs
+    - downloads
+python:
+  - 2.7.13
 env:
-  - PLONE_VERSION=4.2
```
Comment (hvelarde) [inline .travis.yml]: if you're not removing support for Plone 4.2 why did removed this line?

it's totally valid to remove support for Plone 4.3 but you need to add that to the changelog.

File: dexterity/membrane/membrane_helpers.py [hunk 2]
```diff
@@ -36,7 +37,7 @@ def get_brains_for_email(context, email, request=None):
     # if request is None:
     #     try:
     #         request = context.REQUEST
-    #     except:
+    #     except:  # noqa
```
Comment (hvelarde) [inline dexterity/membrane/membrane_helpers.py]: this is a bad idea; you have to check for `AttributeError` in the exception.

File: setup.py [hunk 3]
```diff
@@ -36,10 +37,9 @@
     include_package_data=True,
     zip_safe=False,
     install_requires=[
-        'Products.CMFPlone>=4.2',
+        'Products.CMFPlone>=4.3',
```
Comment (hvelarde) [inline setup.py]: you are dropping support for Plone 4.2 explicitly here.

File: .travis.yml [hunk 4]
```diff
@@ -1,23 +1,25 @@
...
-  - buildout -c travis.cfg annotate
-  - buildout -c travis.cfg -N -q -t 3
-before_script:
-  - export DISPLAY=:99.0
-  - sh -e /etc/init.d/xvfb start
-script: bin/test
+  - virtualenv .
+  - bin/pip install -r requirements.txt
+  - sed -ie "s#plone-5.1.x.cfg#plone-$PLONE_VERSION.x.cfg#" travis.cfg
+  - bin/buildout -c travis.cfg buildout:download-cache=downloads code-analysis:return-status-codes=True
```
Comment (hvelarde) [inline .travis.yml]: please add a `bin/buildout annotate` above he so we can check that the right versions are being tested.

**Comment by agitator** [comment]
Sorry if I was unclear...
I dropped support for 4.2, since I don't wanted to setup tests for 4.2, the new versions where hassle enough.
I added a 1.2.x branch for fixes

**Comment by jensens** [comment]
+1 for a legacy branch and pushing master forward

**Comment by hvelarde** [comment]
and that's exactly why fixing the builds, code analysis and so on should be done outside this PR, to leave the legacy branch on a cleaner state.

**Pull request merged**